### ISSUE TITLE
Mandatory hero of the same race at the start of new week

### ISSUE THREAD

**Issue by alo23nso** [ihhub/fheroes2#8303]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

You are always given hero of the same race you started the game at the beggining of every week in OG. In fheroes2 you get random race heroes.

### Save file

Very easy to check.

### Additional info
Fheroes2 has only native hero at the start of the game (1st week)

**Comment by alo23nso** [comment]
I've checked this save and come with some concusions.
 All available necromancers are already >1lvl. (there is no Sandro since its not prince of loyalty)
 I've made couple of loads also trying to change "seed" somehow and i've got strange results.
 Sometimes ive got only necromancers in tavern for many many weeks and no other class.
 Sometimes ive got none of necromancers for couple weeks and no necromancer.
 Sometimes ive got mixed class, but with rather dominance of necro. (also this was the very rarest scenario)
